storehaus-postgresql/src/main/scala/com/twitter/storehaus/postgres/PostgresStore.scala

```diff
@@ -0,0 +1,123 @@
+package com.twitter.storehaus.postgres
+
+import com.twitter.storehaus.Store
+import com.twitter.util.{Future, Time}
+import roc.postgresql.{Client, Element, Request, Result}
+
+/**
+  * Key value storage with PostgreSQL (>9.5) as backend
+  *
+  * Created by ponkin on 9/24/16.
+  */
+object PostgresStore {
+
+  def apply[K: Show : Read, V: Show : Read](client: Client, table: String, kCol: String, vCol: String) =
+    new PostgresStore[K, V](client, table, kCol, vCol)
+}
+
+class PostgresStore[K: Show : Read, V: Show : Read]
+(protected[postgres] val client: Client, table: String, kCol: String, vCol: String)
+  extends Store[K, V] {
+
+  implicit def toPostgresValue(e: Element): PostgresValue = RocPostgresValue(e)
+
+  private val EMPTY_STRING = ""
+
+  protected val SELECT_SQL_PREFIX =
+    s"SELECT $kCol, $vCol FROM $table WHERE $kCol"
+
+  protected val DELETE_SQL_PREFIX =
+    s"DELETE FROM $table WHERE $kCol"
+
+  override def get(key: K): Future[Option[V]] = {
+    val result = run(selectSQL(List(key)))
+    result.map {
+      _.headOption.map(col => Read[V].read(col.get(Symbol(vCol))))
+    }
+  }
+
+  override def multiGet[K1 <: K](ks: Set[K1]): Map[K1, Future[Option[V]]] = {
+    if (ks.isEmpty) {
+      Map.empty
+    } else {
+      val result = run(selectSQL(ks.toList))
+      val reqMap = result.map {
+        _.map(row => Read[K].read(row.get(Symbol(kCol))) -> Read[V].read(row.get(Symbol(vCol)))).toMap
+      }
+      ks.foldLeft(Map.empty[K1, Future[Option[V]]]) { (acc, key) =>
+        acc + (key -> reqMap.map(_.get(key)))
+      }
```

Comment:
```scala
ks.iterator.map { key =>
  (key, reqMap.map(_.get(key)))
}.toMap
```
would be more direct than building the map yourself (and maybe faster since it can use a MapBuilder).